cmd/soroban-rpc/lib/preflight.h

```diff
@@ -15,6 +15,7 @@ typedef struct CPreflightResult {
     char *error; // Error string in case of error, otherwise null
     char *result; // SCVal XDR in base64
     char *preflight; // LedgerFootprint XDR in base64
+    char **auth; // NULL terminated array of XDR ContractAuths in base64
```

Comment:
given that you're packing on the auths in a single contagious array, I think that the "correct" datatype here is `char*` rather then `char**`.

Reply:
I am not packing the auths in a contiguous array.

Reply:
sorry - my bad; I wasn't reading this correctly.

Reply:
See https://github.com/stellar/soroban-tools/pull/395#discussion_r1097685983 in case it helps.

Reply:
It follows the same format as `**char argv` in C

Reply:
aiui it now, they're not actually packed in a single contiguous array. This main array is an array of pointers to the individual arrays of each string.